pyEpiabm/pyEpiabm/sweep/update_place_sweep.py

```diff
@@ -24,25 +28,104 @@ def __call__(self, time: float):
+                    place.empty_place(groups_to_empty=[-1])
+                    candidate_list = [person for person in place.cell.persons
+                                      if person not in place.persons]
+                    self.update_place_group(place, group_index=-1,
+                                            mean_capacity=mean_cap,
+                                            person_list=candidate_list)
+
+                elif place.place_type.name == "OutdoorSpace":
+                    place.empty_place()
+                    self.update_place_group(place)
+
+    def update_place_group(self, place, mean_capacity: float = 25,
+                           power_law_params: list = None,
+                           group_index: int = None,
+                           group_size: int = 0, person_list: list = None,
+                           person_weights: list = None):
+        """Specific method to update people in a place or place group.
+
+        Parameters
+        ----------
+        place : Place
+            Place to change
+        mean_capacity : int
+            Average number people of in this place
+        power_law_params : list
+            Should only be given for workplaces, contains the further
+            parameters used to calculate place_size. List ent,ries should
+            take the order [Maximum size, Offset, Power]
+        group_index : int
+            If specified, the index of the group to be added to
+        group_size: int
+            Average size of the groups in each place
+        person_list: list
+            List of people that may be present in the place
+        person_weights : list
+            Weights for people in list
+        """
+        # If a specific list of people is not provided, use the whole cell
+        if person_list is None:
+            person_list = place.cell.persons
+
+        # Ensure that the number of people put in the place
+        # is at most its capacity or the total number of
+        # people in the cell. Will use a power law calculation if
+        # parameters are provided, and a Poisson distribution is not.
+        if power_law_params is None:
+            new_capacity = np.random.poisson(mean_capacity)
+        else:
+            assert len(power_law_params) == 3, \
+                "Incorrect number of power law parameters given"
+            [maximum, offset, power] = power_law_params
+            s = (offset / (offset + maximum - 1)) ** power
+            r = random.random()
+            num = offset * ((1 - s) * r + s) ** (-1 / power) + 1 - offset
+            new_capacity = math.floor(num)
+
+        new_capacity = min(new_capacity, len(person_list))
+
+        if len(person_list) <= 0:
+            logging.warning("No people in the person list supplied.")
+        count = 0
+
+        try:
+            num_groups = np.random.poisson(math.ceil(new_capacity/group_size))
+        except ZeroDivisionError:
```

Comment:
Can you comment why this might occur? I assume it's when `group_size = 0` but not really sure why the function would be called with that, and why we'd want to handle it in this way

Reply:
In the outdoor spaces or other places where there aren't groups then want trivially there to be one group in that place which is caught in the try/except